dune/iga/ibraGeometry.hh

```diff
@@ -0,0 +1,369 @@
+//
+// Created by Henri on 12.03.2023.
+//
+
+#ifndef DUNE_IGA_IBRAGEOMETRY_HH
+#define DUNE_IGA_IBRAGEOMETRY_HH
+
+#include "nurbspatchdata.hh"
+
+#include <nlohmann/json.hpp>
+
+namespace Dune::IGA::Ibra {
+
+  enum Type {
+    NurbsCurveGeometry2D,
+    NurbsSurfaceGeometry3D,
+    BrepLoopType,
+    BrepTrimType,
+    BrepType,
+    NoType
+  };
+
+  Type typeForTypeString(const std::string& typeString) {
+    if (typeString == "NurbsCurveGeometry2D")
+      return Type::NurbsCurveGeometry2D;
+    else if (typeString == "NurbsSurfaceGeometry3D")
+      return Type::NurbsSurfaceGeometry3D;
+    else if (typeString == "BrepLoop")
+      return Type::BrepLoopType;
+    else if (typeString == "BrepTrim")
+      return Type::BrepTrimType;
+    else if (typeString == "Brep")
+      return Type::BrepType;
+    else
+      return Type::NoType;
+  }
+
+  class Geometry {
```

Comment:
I suggest joining `Curve` and `Surface` via template.
E.g.
```cpp
struct GeometryBase {
 std::string typeString;
    std::string key;
    Type type{Type::NoType};
};

template <int dim>
class Geometry
{
   class Geometry {
    std::array<int, 2> degree{};
    std::array<int, 2> n_controlPoints{};
    std::array<std::vector<double>, 2> knots;
    Dune::DynamicMatrix<double> controlPoints;

    std::vector<std::vector<typename NURBSPatchData<dim, outputDim>::ControlPointType>> compileControlPoints() {

}
...

Then you can still use

```cpp
using Curve = Geometry<1>;
using Surface = Geometry<1>;
...

you can also think about to make the new Geometry class a new template not only of the geometry dimension but also of the embedding space.

This would be more in line with the other geometry classes.

But maybe here the naming is a bit wrong, since it is not doing geometry in the dune sense.
You only read in controlpoints and knotvectors from json to the internal formats Therefore you can choose a name like `JsonNurbsData`and then you have a `transform`function instead of compiling.